### PR TITLE
Remove the notification specialized class switcher

### PR DESCRIPTION
**Pull request by danidoni**
Depends on #16477

This should be merged after #16477 is deployed on our internal instances

### CODE REVIEW

**Comment by saraycp** [comment]
Rebased on master

**Comment by danidoni** [comment]
All notifications have been migrated as STI. So this PR is free to be reviewed and merged

File: src/api/.rubocop_todo.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
 # This configuration was generated by
 # `rubocop --auto-gen-config --auto-gen-only-exclude --exclude-limit 100`
-# on 2024-07-15 11:49:40 UTC using RuboCop version 1.65.0.
+# on 2024-07-15 16:14:45 UTC using RuboCop version 1.65.0.
```
Comment (eduardoj) [inline src/api/.rubocop_todo.yml:3]: If we only update the `Offense count` of some cops, we don't need to update this file.

File: src/api/spec/factories/notification.rb [hunk 2]
```diff
@@ -23,104 +32,108 @@
...
       end
     end
 
-    trait :cleared_decision do
-      event_type { 'Event::ClearedDecision' }
-      notifiable { association(:decision_cleared) }
+    factory :notification_for_report, class: 'NotificationReport' do
+      trait :create_report do
+        event_type { 'Event::CreateReport' }
+        type { 'NotificationReport' }
```
Comment (eduardoj) [inline src/api/spec/factories/notification.rb:102]:
```suggestion
    factory :notification_for_report, class: 'NotificationReport' do
      trait :create_report do
        event_type { 'Event::CreateReport' }
```
The `type { 'NotificationReport' }` line looks redundant to me. In other sections of this same file, defining a `factory` with a `class` attribute is enough, and does not include a `type` line.

**Review by eduardoj — APPROVED** [review]

**Pull request merged**